sale_order_invoice_qty_number/models/sale_order.py

```diff
@@ -0,0 +1,14 @@
+# © 2023 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3 - See http://www.gnu.org/licenses/lgpl-3.0.html
+from odoo import models, fields, api
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    invoice_ids = fields.Many2many("account.move")
+    invoice_number_count = fields.Integer(compute= '_compute_invoice_number_count', string='Invoice count', store= True)
+
+    @api.depends('order_line.invoice_lines')
+    def _compute_invoice_number_count(self):
+        for invoice in self:
```

Comment:
This recordset is for sale orders, that variable name could lead to confusion:
```suggestion
        for order in self:
```